libagent/gpg/keyring.py

```diff
@@ -29,19 +34,23 @@ def get_agent_sock_path(env=None, sp=subprocess):
     """Parse gpgconf output to find out GPG agent UNIX socket path."""
     args = [util.which('gpgconf'), '--list-dirs']
     output = check_output(args=args, env=env, sp=sp)
-    lines = output.strip().split(b'\n')
-    dirs = dict(line.split(b':', 1) for line in lines)
+    lines = re.split('[\n\r]+', output.decode('utf-8'))
+    dirs = {k: urllib.parse.unquote(v).encode() for k, v in (
```

Comment:
Why `urllib.parse.unquote` is needed here (and below)?

Reply:
Because the components are URL-encoded. Notably, paths which contain `:`, e.g. `C:\\Program Files (x86)\\GnuPG\\bin\\gpg.exe` are encoded as `C%3a\\Program Files (x86)\\GnuPG\\bin\\gpg.exe`